contribution/translation/prevalidator.php

```diff
@@ -67,282 +63,78 @@ public function get_helper()
+		// Rename the extracted directory to be the ISO code.
+		$finder = new Finder();
+		$iterator = $finder->directories()->in($path)->depth('== 0');
 
-		// Now we have all array keys... let us have a look within $contents if all keys are there...
-		// This can take a bit because we check every key... luckily with strpos...
-		foreach (array_keys($lang) as $current_key)
+		if ($iterator->count() == 1)
 		{
-			if (is_string($current_key) && !is_array($lang[$current_key]) && strpos($file_data, "'{$current_key}'=>") === false)
-			{
-				if (empty($lang[$current_key]))
-				{
-					continue;
-				}
+			$iterator = $finder->getIterator();
+			$iterator->rewind();
 
-				$missing_keys[] = $current_key;
-			}
+			// We know there's only one result
+			$root_extracted = $iterator->current();
 		}
 
-		return $missing_keys;
-	}
+		// Rename the directory to use the iso code
+		$new_directory_name = str_replace($root_extracted->getFilename(), $origin_iso, $root_extracted->getPathname());
+		$file_system = new Filesystem();
+		$file_system->rename($root_extracted->getPathname(), $new_directory_name);
 
-	private function add_license_files($license_file, $reference_path)
-	{
-		$res = fopen($license_file, 'w');
-		fwrite($res, file_get_contents($reference_path . '/docs/COPYING'));
-		fclose($res);
-	}
+		// Get the British English language in there too
+		$en_path = $this->get_helper()->get_root_path() . 'includes/language_packages/' . self::BRITISH_ENGLISH . '.zip';
+		$zip = new \ZipArchive();
+		$result = $zip->open($en_path);
 
-	private function add_htm_files($lang_root_path, $reference_path)
-	{
-		$htm_files = array('', 'acp/', 'mods/');
+		// TODO: is there a better way to handle the en package?
 
-		if (!is_dir($lang_root_path . 'mods'))
+		if ($result)
 		{
-			mkdir($lang_root_path . 'mods');
-			phpbb_chmod($lang_root_path . 'mods', CHMOD_READ | CHMOD_WRITE);
-		}
+			// Unzip the revision to a temporary folder
+			$zip->extractTo($path);
+			$zip->close();
 
-		foreach ($htm_files as $htm_file)
-		{
-			$res = fopen($lang_root_path . $htm_file . 'index.htm', 'w');
-			fwrite($res, file_get_contents($reference_path . '/language/en/index.htm'));
-			fclose($res);
+			// Change to "en"
+			$file_system->rename(sprintf('%s/%s', $path, self::BRITISH_ENGLISH), sprintf('%s/%s', $path, self::EN));
 		}
-	}
 
-	/**
-	 * Basically flattens the files from all subdirectories of $root_dir into an array
-	 *
-	 * @param string $root_dir
-	 * @param string $dir DO NOT USE, recursive!
-	 * @return array
-	 */
-	private function lang_filelist($root_dir, $dir = '')
-	{
-		clearstatcache();
-		$matches = array();
-
-		// Add closing / if present
-		$root_dir = ($root_dir && substr($root_dir, -1) != '/') ? $root_dir . '/' : $root_dir;
+		// Parameters for the validation script
+		$inputs = array(
+			// Arguments
+			'command' 			=> 'validate',
+			'origin-iso' 		=> $origin_iso,
+
+			// Options
+			'--phpbb-version' 	=> '3.2',
```

Comment:
What happens in future versions then? Perhaps this can be a parameter?

Reply:
Yeah it can, we can hook it into the using phpBB version number selected during the language pack revision submission.